src/acq/adc.c

```diff
@@ -198,6 +199,7 @@ static void bl_acq_adc_group_disable(bl_acq_adc_group_t *adc_group)
 typedef struct
 {
 	unsigned channel_count;
+	uint8_t  src_mask[ADC_CHANNEL_COUNT];
```

Comment:
I think I'd add a `bool src_is_pd[ADC_CHANNEL_COUNT];` here too.

And also add a:

```c
/**
 * Determine whether a given source is a photodiode
 *
 * \param[in]  src  The source to check.
 * \return True if the source is a photodiode, false otherwise.
 */
static inline bool bl_acq_source_is_photodiode(enum bl_acq_source src)
{
	return (src >= BL_ACQ_PD1) && (src <= BL_ACQ_PD4);
}

to `src/acq.h`.

Reply:
+1